raft/src/main/java/org/apache/kafka/snapshot/Snapshots.java

```diff
@@ -68,15 +68,22 @@ public static Path snapshotPath(Path logDir, OffsetAndEpoch snapshotId) {
         return snapshotDir(logDir).resolve(filenameFromSnapshotId(snapshotId) + SUFFIX);
     }
 
-    public static Path createTempFile(Path logDir, OffsetAndEpoch snapshotId) throws IOException {
+    public static Path createTempFile(Path logDir, OffsetAndEpoch snapshotId) {
         Path dir = snapshotDir(logDir);
+        Path tempFile;
 
-        // Create the snapshot directory if it doesn't exists
-        Files.createDirectories(dir);
-
-        String prefix = String.format("%s-", filenameFromSnapshotId(snapshotId));
+        try {
+            // Create the snapshot directory if it doesn't exists
+            Files.createDirectories(dir);
 
-        return Files.createTempFile(dir, prefix, PARTIAL_SUFFIX);
+            String prefix = String.format("%s-", filenameFromSnapshotId(snapshotId));
+            tempFile = Files.createTempFile(dir, prefix, PARTIAL_SUFFIX);
```

Comment:
nit: couldn't we return here and remove `tempFile`?

Reply:
You're right！Thanks